### DISCUSSION TITLE
Is it possible to trigger a capture from a templater's template ?

### DISCUSSION THREAD

**Discussion by lyndhurst** [chhoumann/quickadd#471] [General]
I read about macros, format syntax and the quickadd api, but I am not sure I found what I was looking for.

I would simply like to "automagically" trigger a QuickAdd capture at the end of a Templater's template execution.

Does anyone know how to achieve this ?

**Comment by chhoumann** [comment]
You can execute any Obsidian command with JavaScript, which Templater supports. 

I can't test it right now, but going from memory it's something like:
`app.commands`

Which is an object where you can find the various Obsidian commands - and execute them. 
Binding a Capture to a command, with the lightning ⚡️ symbol, will let you run it like that. 

Besides this, I'd have to know the entire workflow to help further😁

**Reply by FelipeRearden** [reply]
Hello @chhoumann  and @lyndhurst 

> I can't test it right now, but going from memory it's something like: `app.commands`

@chhoumann  this is the command `app.commands.executeCommandById('ID')`

@lyndhurst 
To get the ID for the QuickAdd Choice you can use this Templater script from the link below to:

https://discord.com/channels/686053708261228577/840286238928797736/1095503621250089001

It will open a suggester with all commands from the vault and return the syntax to execute then in a templater script. I sue it all the time to add QuickAdd choice to my templater scripts

I hope it helps 🙏

**Comment by lyndhurst** [comment]
Hi @chhoumann  , @FelipeRearden ,
Thanks guys for the tip, it actually opens a lot of possibilities. I cannot test it before monday, maybe tomorrow... I am gonna look into it, and make sure to update with thread with my progress, it might help someone else :)
Thanks again, and have a good week end :+1:

**Comment by lyndhurst** [comment]
It took me longer than I expected, but I managed to test adding "capture triggers" in my templates, and it works exactly as described by @chhoumann and @FelipeRearden . Ultimately I just added a line that looks like this:

```js
await app.commands.executeCommandById('quickadd:choice:70010db8-1d85-47d3-bc4f-b3151fa721f3')
```

The only problem I encountered is that I could not access the link provided above by @FelipeRearden to a discord channel (to be honest I did not insist much). I still could retrieve the command ID using the trick provided by the `obsidian-vimrc-support` plugin author described [here](https://github.com/esm7/obsidian-vimrc-support#executing-obsidian-commands-with-obcommand).

To summarize, if you have the plugin installed (or vim features enabled in settings, I am not sure), just issue a command exactly like in `vim` by typing `:obcommand <Enter>`. Nothing happens at this point, but if you open the developer console hitting `Ctrl-Shift-I`, all available obsidian commands will be printed out there... Then it is just a matter of finding the one you are looking for in the list.

For QuickAdd custom commands, it can be a little tricky if you have a lot of them (God knows that I do !) because they are named using UUIDs, not the arbitrary capture name, so it mlght require a little trial and error to identify the correct one. As always in life, there is an upside to that naming style; you do not have to go over all your references to a command if you simply rename it.

Thanks a lot guys for answering me so quickly, your help is greatly appreciated :+1:

**Comment by FelipeRearden** [comment]
Hello @lyndhurst 

> The only problem I encountered is that I could not access the link provided above by @FelipeRearden to a discord channel (to be honest I did not insist much).

This is the script that I was talking about: 


```
<%*
const cmd = await tp.system.suggester(cmd => cmd.name, Object.values(this.app.commands.commands), false, "", 10)
await navigator.clipboard.writeText(`app.commands.executeCommandById('${cmd.id}')`)
%>
```

I wish you a fantastic  day!

**Reply by lyndhurst** [reply]
Cheers, a great day to you too.

**Comment by chhoumann** [comment]
Closing this one as resolved. For anyone finding this today, there's a nicer route now: QuickAdd's API can run a choice by name, so no UUID hunting:

```js
await app.plugins.plugins.quickadd.api.executeChoice("My capture");
```

Docs: https://quickadd.obsidian.guide/docs/QuickAddAPI
